rosenpass/src/protocol.rs

```diff
@@ -754,7 +852,127 @@ pub struct HandleMsgResult {
 }
 
 impl CryptoServer {
-    /// Respond to an incoming message
+    /// Process a message under load
+    /// This is one of the main entry point for the protocol.
+    /// Keeps track of messages processed, and qualifies messages using
+    /// cookie based DoS mitigation. Dispatches message for further processing
+    /// to `process_msg` handler if cookie is valid, otherwise sends a cookie reply
+    /// message for sender to process and verify for messages part of the handshake phase
+    /// (i.e. InitHello, InitConf messages only). Bails on messages sent by responder and
+    /// non-handshake messages.
+    pub fn handle_msg_under_load(
+        &mut self,
+        rx_buf: &[u8],
+        tx_buf: &mut [u8],
+        peer: PeerPtr,
+        ip_addr_port: &[u8],
+    ) -> Result<HandleMsgResult> {
+        let mut cookie_value = [0u8; 16];
+        match self.cookie_secret.lifecycle(&self) {
```

Comment:
Can we avoid the code duplication and use the same code as in active_biscuit_key?